src/riot-rs-threads/src/lib.rs

```diff
@@ -118,11 +122,16 @@ impl Threads {
         &mut self.threads[thread_id as usize]
     }
 
-    /// Returns an unused ThreadId / Thread slot.
+    /// Returns an unused [`ThreadId`] / [`Thread`] slot.
     fn get_unused(&mut self) -> Option<(&mut Thread, ThreadId)> {
         for i in 0..THREADS_NUMOF {
             if self.threads[i].state == ThreadState::Invalid {
-                return Some((&mut self.threads[i], i as ThreadId));
+                #[allow(
+                    clippy::cast_possible_truncation,
+                    reason = "all iteration indices are valid `ThreadId`s"
```

Comment:
This might be a nice opportunity for refactoring threads to provide an iterator with typed `.enumerate()`.